RetroAchievements/RAWeb
# feat(home): migrate Active Players to React

This is PR 3 of the React home page migration, and a follow-up to https://github.com/RetroAchievements/RAWeb/pull/2802. These are the planned steps of the migration:

~PR 1 of 4: Build some basic components with hardcoded data. Lots of functionality missing, no back-end.~
~PR 2 of 4: Use real data from the back-end / Laravel controllers rather than hardcoded data.~
**PR 3 of 4: Rebuild the Active Players component.**
PR 4 of 4: Finalize the swap of the PHP home page for the React home page.

This PR sets up the Active Players component on `/demo/home` with minor behavior changes.

### How to test this PR
1. Open _LoadThinActivePlayersListAction.php_.
2. Adjust `$lookbackMinutes` on line 25 from 10 to something like 28000.
3. Visit `/demo/home` and observe that active players and trending games are populated. If you don't see any, keep increasing the `$lookbackMinutes` number.

**Before**
![Screenshot 2024-11-19 at 4 29 03 PM](https://github.com/user-attachments/assets/6a6ecc42-8df1-4dc7-b3f8-e2382d3c9e8f)


**After**
![Screenshot 2024-11-19 at 4 28 33 PM](https://github.com/user-attachments/assets/029f8bd7-f497-40f6-8688-c61109b14e30)


Note: Player counts in the screenshots above are artificially inflated due to my high lookback time for testing purposes.

There are a few minor behavior changes to note:
* Game titles are now the focal point of each row, and RP is slightly suppressed.
* The "Menu" button now uses a search icon.
* The "last updated" label has been removed. This updates so infrequently I doubt most users even realize it happens automatically if the page is idle.
* The client-side fetch no longer fetches all records. In prod, this slows down the page considerably as this action is immediately followed by inserting thousands of DOM nodes onto the page. The UI now fetches in increments of 300.
* Searching by system is no longer possible.

The UI has 100% code coverage.
